src/renderers/native/ReactNative.js

```diff
@@ -21,13 +21,15 @@ var ReactUpdates = require('ReactUpdates');
 
 var findNodeHandle = require('findNodeHandle');
 
+import type { ReactElement } from 'ReactElementType';
+
 ReactNativeDefaultInjection.inject();
 
 var render = function(
-  element: ReactElement<any>,
+  element: ReactElement | string | number | null | false,
```

Comment:
I don't think anything other than `ReactElement` is valid for render right now. Possibly `null`. I.e. the same things that are valid to return from `render`.

I don't think we can use the internal `ReactElement` type for the public API since the public API [built into Flow](https://github.com/facebook/flow/blob/master/lib/react.js#L121-L126) doesn't have [the extra hidden fields](https://github.com/facebook/react/blob/master/src/isomorphic/classic/element/ReactElementType.js#L28-L36) that we use internally.

So anyone passing Flow's ReactElement into this function would get an error since it is missing the extra fields.

Additionally, we'd lose the extra information Flow keeps using the generic parameterization.